packages/clay-shared/src/setElementFullHeight.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * SPDX-FileCopyrightText: © 2021 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * Adds a height style that triggers a transition when collapsed
+ */
+export function setElementFullHeight(element: HTMLElement) {
+	if (element) {
+		// Cloned into a new array since `.reduce` is not a method on an HTMLCollection
+		const height = Array.prototype.slice
+			.call(element.children)
+			.reduce(
+				(acc: number, child: HTMLElement) => acc + child.clientHeight,
+				0
+			);
```

Comment:
I don't think I'd create a new object only to have it immediately GC'ed like this, especially if it involves me having to use a relatively esoteric `slice` + `call`. A simple `for` loop would do and arguably be more readable (no need for type annotations, no need for `reduce`, which some people find hard to read etc):

```js
let height = 0;

for (let i = 0; i < element.children.length; i++) {
    height += element.children[i].clientHeight;
}
```